v2/AdminCore.Services/EventService.cs

```diff
@@ -0,0 +1,404 @@
+﻿using AdminCore.Common.Interfaces;
+using AdminCore.Common.Message;
+using AdminCore.Common.Message.Elements;
+using AdminCore.Constants.Enums;
+using AdminCore.DAL;
+using AdminCore.DAL.Models;
+using AdminCore.DTOs.Event;
+using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AdminCore.Services
+{
+  public class EventService : IEventService
+  {
+    private readonly IDatabaseContext _databaseContext;
+
+    private readonly IMapper _mapper;
+
+    public EventService(IDatabaseContext databaseContext, IMapper mapper)
+    {
+      _databaseContext = databaseContext;
+      _mapper = mapper;
+    }
+
+    public ResponseMessage<IList<EventDto>> GetAnnualLeaveByEmployee(int employeeId)
+    {
+      ResponseMessage<IList<EventDto>> responseMessage = new ResponseMessage<IList<EventDto>>(null);
+      var annualLeave = _databaseContext.EventRepository.Get(x =>
+        x.EventType.EventTypeId == (int)EventTypes.AnnualLeave
+        && x.Employee.EmployeeId == employeeId);
+      if (!annualLeave.Any())
+      {
+        responseMessage.Payload = _mapper.Map<IList<EventDto>>(annualLeave);
+        responseMessage.Status = MessageConstants.MsgStatusSuccess;
+      }
+      else
+      {
+        responseMessage.Status = MessageConstants.MsgStatusNoRecords;
+      }
+      return responseMessage;
+    }
+
+    public ResponseMessage<IList<EventDto>> GetByDateBetween(DateTime startDate, DateTime endDate, EventTypes eventType)
+    {
+      ResponseMessage<IList<EventDto>> responseMessage = new ResponseMessage<IList<EventDto>>(null);
+      EventDateDto eventDateDto = new EventDateDto
+      {
+        StartDate = startDate,
+        EndDate = endDate
+      };
```

Comment:
Do you need this, seems like you're only using it in your Get, why not just use:

databaseContext.EventDatesRepository.Get(x => x.StartDate >= startDate
                                                                              && x.EndDate <= endDate
                                                                              && x.Event.EventTypeId ==
                                                                              (int)EventTypes.AnnualLeave);